src/panels/CharmsPanel/CharmsPanel.tsx

```diff
@@ -47,16 +47,33 @@ export default function CharmsPanel({
       loading={isLoading}
     >
       <form onSubmit={handleSubmit}>
-        {charms.map((charm) => (
-          <div key={charm.url} className="p-form__group">
-            <RadioInput
-              id={charm.url}
-              label={`${charm.meta?.name} (rev: ${charm.revision})`}
-              checked={selectedCharm === charm.url}
-              onChange={() => setSelectedCharm(charm.url)}
-            />
-          </div>
-        ))}
+        {charms.map((charm) => {
+          const hasActionData =
+            !!charm?.actions?.specs &&
+            !!Object.keys(charm.actions.specs).length;
+          return (
+            <div key={charm.url} className="p-form__group">
+              {hasActionData ? (
+                <RadioInput
+                  id={charm.url}
+                  label={`${charm.meta?.name} (rev: ${charm.revision})`}
+                  checked={selectedCharm === charm.url}
+                  onChange={() => setSelectedCharm(charm.url)}
+                />
```

Comment:
We reuse almost the same component twice here, so I am not sure, but maybe making a component `CharmRadioInput` might improve the code?